mlflow/models/evaluation/default_evaluator.py

```diff
@@ -1015,26 +1015,72 @@ def evaluate(
 
-            inferred_model_type = _infer_model_type_by_labels(self.y)
-
-            if inferred_model_type is not None and model_type != inferred_model_type:
-                _logger.warning(
-                    f"According to the evaluation dataset label values, the model type looks like "
-                    f"{inferred_model_type}, but you specified model type {model_type}. Please "
-                    f"verify that you set the `model_type` and `dataset` arguments correctly."
+            if self.model_type not in ["classifier", "regressor"]:
+                raise MlflowException(
+                    message=f"Unsupported model type {self.model_type}",
+                    erorr_code=INVALID_PARAMETER_VALUE,
                 )
-
             with mlflow.utils.autologging_utils.disable_autologging():
-                if model_type == "classifier":
-                    return self._evaluate_classifier()
-                elif model_type == "regressor":
-                    return self._evaluate_regressor()
-                else:
-                    raise ValueError(f"Unsupported model type {model_type}")
+                self._generate_model_predictions()
+                self._compute_builtin_metrics()
+                self._evaluate_custom_metrics_and_log_produced_artifacts(
+                    log_to_mlflow_tracking=not is_baseline_model
+                )
+                if not is_baseline_model:
+                    self._log_metrics_and_artifacts()
+                return EvaluationResult(metrics=self.metrics, artifacts=self.artifacts)
+
+    def evaluate(
+        self,
+        *,
+        model: "mlflow.pyfunc.PyFuncModel",
+        model_type,
+        dataset,
+        run_id,
+        evaluator_config,
+        custom_metrics=None,
+        baseline_model=None,
+        **kwargs,
+    ):
+        self.dataset = dataset
+        self.run_id = run_id
+        self.model_type = model_type
+        self.evaluator_config = evaluator_config
+        self.dataset_name = dataset.name
+        self.feature_names = dataset.feature_names
+        self.custom_metrics = custom_metrics
+        self.y = dataset.labels_data
+
+        inferred_model_type = _infer_model_type_by_labels(self.y)
+
+        if inferred_model_type is not None and model_type != inferred_model_type:
+            _logger.warning(
+                f"According to the evaluation dataset label values, the model type looks like "
+                f"{inferred_model_type}, but you specified model type {model_type}. Please "
+                f"verify that you set the `model_type` and `dataset` arguments correctly."
+            )
+
+        if evaluator_config.get("_disable_candidate_model", False):
+            evaluation_result = EvaluationResult(metrics=dict(), artifacts=dict())
+        else:
+            evaluation_result = self._evaluate(
+                model, is_baseline_model=evaluator_config.get("_disable_candidate_model", False)
+            )
```

Comment:
Nit: I think the config get is no longer necessary here?
```suggestion
            evaluation_result = self._evaluate(model, is_baseline_model=False)
```